libs/platform/src/lib/table/table.component.ts

```diff
@@ -1810,6 +1810,7 @@ export class TableComponent<T = any> extends Table<T> implements AfterViewInit,
     /** @hidden */
     private _listenToColumns(): void {
         this.columns.changes.pipe(startWith(null)).subscribe(() => {
+            this.setTableState({ ...this.getTableState(), columns: this.columns.map((column) => column.name) });
```

Comment:
Isn't this mean that all other columns (that were previously hidden with table state) will also appear?

Reply:
Where should we set this state update function in your opinion ? @N1XUS

Reply:
I think we need to compare previous array of columns with new one. If new array one has added columns, push them to the state and ignore updating previous one.
This would be better due to the fact that users can manually hide some columns via p13 dialog, and when this check triggers, that config will be useless.